Realm/ObjectStore/object_store.cpp

```diff
@@ -479,3 +476,106 @@ void ObjectStore::delete_data_for_object(Group *group, const StringData &object_
     }
 }
 
+
+
+InvalidSchemaVersionException::InvalidSchemaVersionException(uint64_t old_version, uint64_t new_version) :
+    m_old_version(old_version), m_new_version(new_version)
+{
+    m_what = "Provided schema version " + std::to_string(old_version) + " is less than last set version " + std::to_string(new_version) + ".";
+}
+
+DuplicatePrimaryKeyValueException::DuplicatePrimaryKeyValueException(std::string object_type, Property const& property) :
```

Comment:
The complex value types (`std::string`, `std::vector`) in these constructor arguments should probably all be taken as const references.